src/node_url.cc

```diff
@@ -787,11 +787,13 @@ bool ToASCII(const std::string& input, std::string* output) {
 #else
 // Intentional non-ops if ICU is not present.
 bool ToUnicode(const std::string& input, std::string* output) {
+  std::cerr << "Warning: Cannot convert to unicode when intl is disabled!\n";
```

Comment:
These definitely would need to use the `emitWarning()` mechanism. Also, I'd prefer if these were opt-in and gated by a command line flag.

Reply:
> It would be nice if these warnings were emitted either via `process.emitWarning()` in JS (it’s okay to create wrapper functions for `--without-intl` builds imo) or `ProcessEmitWarning()` in C++ instead, because that way this works with `--no-warning`, `--trace-warning`, `process.on('warning')`, etc.

Im not familiar with those functions, but it does sound better than just `printf(stderr` …
Does emitwarning elide duplicate warnings? i.e. only print a warning once?

> Also, yes, I think we should treat this as semver-major. Apart from that, I have no strong opinions here (except for those that I already voiced and that have been addressed :)).

I think it is major because it's long standing behavior.

( @aduh95 out of curiosity, just to keep myself up to date, what is the major use case for without-intl- smaller size, and/or embedded devices, ?)

Reply:
> what is the major use case that you see for without-intl?

@srl295 As a matter of fact, I don't use it :) I stumbled upon this weird behavior when working on https://github.com/nodejs/node/pull/35091, so I figure that was probably not intentional.

@jasnell do you know if `emitWarning` API is available from C++-land and how to access it? I agree that is the "correct" way to do it, but I didn't manage to get it working.

Another approach I have thought of would be to call the `punycode` JS module, but again I don't know if that's possible from C++-land. That would remove the need for a warning (or a throw).

Reply:
Yes, do a search in `src` for `ProcessEmitWarning` and you'll see how to make that happen